straxen/analyses/daq_waveforms.py

```diff
@@ -0,0 +1,141 @@
+import numba
+import pandas
+import straxen
+import numpy as np
+import strax
+import pymongo
+import typing
+import matplotlib.pyplot as plt
+
+
+@straxen.mini_analysis()
+def daq_plot(context, figsize=(14, 15), lower_panel_height=6, group_by='link', **kwargs):
+    """
+    Plot with peak, records and records sorted by "link" or "ADC ID"
+    """
+    f, axes = plt.subplots(3, 1,
+                           figsize=figsize,
+                           gridspec_kw={'height_ratios': [1, 1, lower_panel_height]})
+
+    plt.sca(axes[0])
+    plt.title('Peaks')
+    context.plot_peaks(**kwargs,
+                       single_figure=False)
+    xlim = plt.xlim()
+    plt.xticks(rotation=0)
+    plt.grid('y')
+
+    plt.sca(axes[1])
+    plt.title('Recods (by channel number)')
+    context.plot_records_matrix(**kwargs,
+                                single_figure=False)
+    plt.xticks(rotation=0)
+    plt.grid('x')
+    plt.xlim(*xlim)
+
+    plt.sca(axes[2])
+    plt.title('Records (by reader, link, crate)')
+    context.plot_records_matrix(**kwargs,
+                                group_by=group_by,
+                                single_figure=False)
+    plt.xlim(*xlim)
+    plt.grid()
+
+
+def _get_daq_config(
+        context: strax.Context,
+        run_id: str,
+        config_name: str = 'daq_config',
+        run_collection: typing.Optional[pymongo.collection.Collection] = None) -> dict:
+    """
+    Query the runs database for the config of the daq during this run.
+    Either use the context of the runs collection.
+    """
+    if not context.storage[0].__class__.__name__ == 'RunDB' and run_collection is None:
+        raise NotImplementedError('Only works with the runsdatabase')
+    if run_collection is None:
+        run_collection = context.storage[0].collection
+    daq_config = run_collection.find_one({"number": int(run_id)},
+                                         projection={config_name: 1}).get(config_name, None)

```

Comment:
I don't strictly agree with this line. If `find_one` returns None this will throw, because `None.get()` isn't a thing. If the document exists, then the `daq_config` field also has to exist. If we somehow manage to find a malformed rundoc, the fact that this code will error is the least of our worries.

Reply:
I *always* screw this up. Thanks